### PR TITLE
Rename B028 to B907, making it optional/opinionated.

### LINKED ISSUE #329: B028 gives wrong suggestion if variable is not a string
B028 gives wrong suggestion if variable is not a string, for example let's have

```
>>> x = 5
>>> print(f'"{x}"')
"5"
```
but the representation does not include quotes:
```
>>> print(f'{x!r}')
5
```
You have to change the type first like:

```
>>> print(f'{str(x)!r}')
'5'
```
but is this an improvement? And the B028 does not suggest this.

See also https://github.com/inducer/pytools/pull/171

### LINKED ISSUE #331: B028 suggestion is wrong when double quotes are required
The `!r` flag surrounds strings by single quotes. In some cases, for example when constructing SQL commands, single quotes are not valid and the suggestion leads to incorrect results.

One solution would be to only trigger B028 when strings are surrounded by single quotes.

### LINKED ISSUE #332: B028 should suggest underlying functions instead of obscure flags
B028 introduced in #319 should suggest `repr` instead of `!r` for readability, these flags were preserved in f-string only for backward compatibility with `.format()`, from PEP 498 [!s, !r, and !a are redundant](https://peps.python.org/pep-0498/#s-r-and-a-are-redundant)

### PR DESCRIPTION
**Pull request by jakkdl**
Fixes #329 
Fixes #331 
Fixes #332

For a certain definition of "fixes" at least, the check could perhaps be improved to make it less opinionated and more useful, but silencing lots of unnecessary warnings is the first step regardless.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -169,7 +169,7 @@ limitations make it difficult.
 
 **B027**: Empty method in abstract base class, but has no abstract decorator. Consider adding @abstractmethod.
 
-**B028**: Consider replacing ``f"'{foo}'"`` with ``f"{foo!r}"`` which is both easier to read and will escape quotes inside ``foo`` if that would appear. The check tries to filter out any format specs that are invalid together with ``!r``. If you're using other conversion flags then e.g. ``f"'{foo!a}'"`` can be replaced with ``f"{ascii(foo)!r}"``. Not currently implemented for python<3.8 or ``str.format()`` calls.
+**B907**: Consider replacing ``f"'{foo}'"`` with ``f"{foo!r}"`` which is both easier to read and will escape quotes inside ``foo`` if that would appear. The check tries to filter out any format specs that are invalid together with ``!r``. If you're using other conversion flags then e.g. ``f"'{foo!a}'"`` can be replaced with ``f"{ascii(foo)!r}"``. Not currently implemented for python<3.8 or ``str.format()`` calls.
```
[the remark this answers is not in the document]
Comment (cooperlees) [reply README.rst:172]: Agree

**Review by cooperlees — CHANGES_REQUESTED** [review]
Thanks for this. This is one I'm 50/50 on too. Please fix the README and I'll merge and ship asap.

File: README.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (jakkdl) [reply README.rst]: oops, I actually had changed that but hadn't committed/pushed since.

**Review by cooperlees — APPROVED** [review]
LGTM - Thanks! Seems popular people want this.

**Pull request merged**